dotCMS/src/main/java/com/dotcms/rendering/velocity/services/FieldLoader.java

```diff
@@ -0,0 +1,99 @@
+package com.dotcms.rendering.velocity.services;
+
+
+import com.dotcms.contenttype.model.field.ConstantField;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.HiddenField;
+import com.dotcms.rendering.velocity.util.VelocityUtil;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+
+import org.apache.velocity.runtime.resource.ResourceManager;
+
+/**
+ * @author Jason Tesser
+ * @since 1.6.5
+ *
+ */
+public class FieldLoader implements DotLoader {
+	public void invalidate(String fieldInode, String contentletIdent, boolean EDIT_MODE,String filePath) throws DotDataException, DotSecurityException {
+		removeFieldFile(fieldInode, contentletIdent, EDIT_MODE);
+	}
+	
+	public InputStream buildVelocity(String fieldInode, String contentInode, boolean EDIT_MODE,String filePath) throws DotDataException, DotSecurityException {
+		InputStream result;
+		Field field = APILocator.getContentTypeFieldAPI().find(fieldInode);
+		if(!UtilMethods.isSet(field)){
+			Logger.warn(this.getClass(),"Field not found.  Unable to load velocity code");
+			return new ByteArrayInputStream("".toString().getBytes());
+		}
+		ContentletAPI conAPI = APILocator.getContentletAPI();
+		Contentlet content = conAPI.find(contentInode, APILocator.getUserAPI().getSystemUser(), true);
+		if(!UtilMethods.isSet(content)){
+			Logger.warn(this.getClass(),"Content not found.  Unable to load velocity code");
+			return new ByteArrayInputStream("".toString().getBytes());
+		}
+		Object contFieldValueObject = conAPI.getFieldValue(content, field);
+		String contFieldValue = "";
+		
+		
+		
+		if(field instanceof ConstantField || field instanceof HiddenField){
+			contFieldValue = field.values() == null ? "" : field.values();
+		}else{
+			contFieldValue = contFieldValueObject == null ? "" : contFieldValueObject.toString();
+		}
+		
+		if(contFieldValue != null && contFieldValue.endsWith("#")){
+			contFieldValue = contFieldValue.substring(0, contFieldValue.length()-1);
+			contFieldValue += "$esc.h";
+		}
+		
+
+        return writeOutVelocity(filePath, contFieldValue);
+
+
+	}
+	
+	public void removeFieldFile (String fieldInode, String contentInode, boolean EDIT_MODE) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)